### DISCUSSION TITLE
DsgQuery returns a `Flow` failed.

### DISCUSSION THREAD

**Discussion by hantsy** [Netflix/dgs-framework#1075] [Q&A]
In these days, I tried to update Dgs/Kotlin Coroutines example, when I updated the QueryTests. 

And run it, and got the following exception.

```bash 
Can't resolve value (/allPosts) : type mismatch error, expected type LIST got class kotlinx.coroutines.flow.FlowKt__BuildersKt$flowOf$$inlined$unsafeFlow$1
```

The query component is:

```kotlin 
@DgsComponent
class PostsDataFetcher(val postService: PostService) {
    @DgsQuery
    fun allPosts(): Flow<Post> = postService.allPosts()
...
}
```

The testing codes is here, https://github.com/hantsy/spring-graphql-sample/blob/master/dgs-kotlin-co/src/test/kotlin/com/example/demo/QueryTests.kt

I got the similar errors when execute the `allPosts` in the /graphiql web interfaces.

```json
{
  "errors": [
    {
      "message": "Can't resolve value (/allPosts) : type mismatch error, expected type LIST",
      "path": [
        "allPosts"
      ],
      "extensions": {
        "classification": "DataFetchingException"
      }
    }
  ],
  "data": null
}
```

**Comment by srinivasankavitha** [comment]
Currently we only support returning Mono and Flux for data fetchers.

**Reply by hantsy** [reply]
There is a issue add Kotlin Coroutines, https://github.com/Netflix/dgs-framework/pull/472
